src/engine/AddCombinedRowToTable.h

```diff
@@ -320,7 +330,7 @@ class AddCombinedRowToIdTable {
     indexBuffer_.clear();
     optionalIndexBuffer_.clear();
     nextIndex_ = 0;
-    std::invoke(blockwiseCallback_, result);
+    std::invoke(blockwiseCallback_, result, mergedVocab_);
```

Comment:
One thing that we can and have to do here:
If the callback leads to consumption of the buffer (which we can easily detect by the buffer being empty afterwards) we can (almost) clear the local vocab:
If the `flush` is part of a resetting of IdTables (setInput), then we can completely clear ist, if it is an intermediate flush, we have to readd the local vocabs of the currently active inputs. (This is probably best done by an argument to `flush()`.
(You can use the `LocalVocab::numSets()` to detect the case that you have few vocabs anyway and clearing is not worth it, if you want to.)

Reply:
Oh, well we pass a mutable reference here to the callback on purpose, so it can move-out the local vocab, practically erasing it completely. But yeah, you're right it needs to be re-added for the currently active tables